test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+
+		ginkgo.By("verifying pod allocations after resize")
+		verifyPodAllocations(resizedPod, expected, true)
+
+		ginkgo.By("patching pod for resize with memory exceeding resource quota")
+		_, pErrExceedMemory := f.ClientSet.CoreV1().Pods(resizedPod.Namespace).Patch(context.TODO(),
+			resizedPod.Name, types.StrategicMergePatchType, []byte(patchStringExceedMemory), metav1.PatchOptions{})
+		gomega.Expect(pErrExceedMemory).To(gomega.HaveOccurred(), "exceeded quota: %s, requested: memory=350Mi, used: memory=700Mi, limited: memory=800Mi",
+			resourceQuota.Name)
+
+		ginkgo.By("verifying pod patched for resize exceeding memory resource quota remains unchanged")
+		patchedPodExceedMemory, pErrEx2 := podClient.Get(context.TODO(), resizedPod.Name, metav1.GetOptions{})
+		framework.ExpectNoError(pErrEx2, "failed to get pod post exceed memory resize")
+		verifyPodResources(patchedPodExceedMemory, expected)
+		verifyPodAllocations(patchedPodExceedMemory, expected, true)
+
+		ginkgo.By(fmt.Sprintf("patching pod %s for resize with CPU exceeding resource quota", resizedPod.Name))
+		_, pErrExceedCPU := f.ClientSet.CoreV1().Pods(resizedPod.Namespace).Patch(context.TODO(),
+			resizedPod.Name, types.StrategicMergePatchType, []byte(patchStringExceedCPU), metav1.PatchOptions{})
+		gomega.Expect(pErrExceedCPU).To(gomega.HaveOccurred(), "exceeded quota: %s, requested: cpu=200m, used: cpu=700m, limited: cpu=800m",
+			resourceQuota.Name)
+
+		ginkgo.By("verifying pod patched for resize exceeding CPU resource quota remains unchanged")
+		patchedPodExceedCPU, pErrEx1 := podClient.Get(context.TODO(), resizedPod.Name, metav1.GetOptions{})
+		framework.ExpectNoError(pErrEx1, "failed to get pod post exceed CPU resize")
+		verifyPodResources(patchedPodExceedCPU, expected)
+		verifyPodAllocations(patchedPodExceedCPU, expected, true)
+
+		ginkgo.By("deleting pods")
+		delErr1 := e2epod.DeletePodWithWait(ctx, f.ClientSet, newPod1)
+		framework.ExpectNoError(delErr1, "failed to delete pod %s", newPod1.Name)
+		delErr2 := e2epod.DeletePodWithWait(ctx, f.ClientSet, newPod2)
+		framework.ExpectNoError(delErr2, "failed to delete pod %s", newPod2.Name)
+	})
+}
+
+func doPodResizeErrorTests() {
+	f := framework.NewDefaultFramework("pod-resize-errors")
+	var podClient *e2epod.PodClient
+	ginkgo.BeforeEach(func() {
+		podClient = e2epod.NewPodClient(f)
+	})
+
+	type testCase struct {
+		name        string
+		containers  []TestContainerInfo
+		patchString string
+		patchError  string
+		expected    []TestContainerInfo
+	}
+
+	tests := []testCase{
+		{
+			name: "BestEffort pod - try requesting memory, expect error",
+			containers: []TestContainerInfo{
+				{
+					Name: "c1",
+				},
+			},
+			patchString: `{"spec":{"containers":[
+						{"name":"c1", "resources":{"requests":{"memory":"400Mi"}}}
+					]}}`,
+			patchError: "Pod QoS is immutable",
+			expected: []TestContainerInfo{
+				{
+					Name: "c1",
+				},
+			},
+		},
+	}
+
+	for idx := range tests {
```

Comment:
Consider running each test case in a framework namespace. That should mitigate some of the race conditions with reusing pod names, and make it easier to debug specific test case failures.

Reply:
Thanks, yes this sounds a good plan, acknowledged will try it and commit your suggestion.

Reply:
For each test case, pods run in dedicated namespace.  Is this what you mean "in a framework namespace"?

To make it easier to debug added a print out when pods are running, for example
> I0529 13:13:00.437653 119149 pod_resize_test.go:1205] pod pod-resize-test-9674/testpod running
> I0529 13:13:37.663620 119149 pod_resize_test.go:1205] pod pod-resize-test-545/testpod running
> I0529 13:14:14.888804 119149 pod_resize_test.go:1205] pod pod-resize-test-823/testpod running
> I0529 13:14:52.109946 119149 pod_resize_test.go:1205] pod pod-resize-test-6310/testpod running
> I0529 13:15:29.685797 119149 pod_resize_test.go:1205] pod pod-resize-test-9427/testpod running
> I0529 13:16:07.424046 119149 pod_resize_test.go:1205] pod pod-resize-test-8712/testpod running
> I0529 13:16:45.685472 119149 pod_resize_test.go:1205] pod pod-resize-test-2920/testpod running
> I0529 13:17:23.995231 119149 pod_resize_test.go:1205] pod pod-resize-test-8948/testpod running
> I0529 13:18:01.741063 119149 pod_resize_test.go:1205] pod pod-resize-test-868/testpod running
> I0529 13:18:41.401856 119149 pod_resize_test.go:1205] pod pod-resize-test-43/testpod running
> I0529 13:19:18.622012 119149 pod_resize_test.go:1205] pod pod-resize-test-7817/testpod running
> I0529 13:19:26.919633 119149 pod_resize_test.go:1205] pod pod-resize-test-271/testpod running
> I0529 13:20:05.182883 119149 pod_resize_test.go:1205] pod pod-resize-test-4692/testpod running
> I0529 13:20:59.993471 119149 pod_resize_test.go:1205] pod pod-resize-test-6327/testpod running
> I0529 13:21:38.522189 119149 pod_resize_test.go:1205] pod pod-resize-test-1534/testpod running
> I0529 13:22:15.726604 119149 pod_resize_test.go:1205] pod pod-resize-test-4925/testpod running
> I0529 13:22:24.383379 119149 pod_resize_test.go:1205] pod pod-resize-test-8094/testpod running
> I0529 13:23:02.615249 119149 pod_resize_test.go:1205] pod pod-resize-test-8735/testpod running
> I0529 13:23:09.920642 119149 pod_resize_test.go:1205] pod pod-resize-test-3210/testpod running
> I0529 13:23:47.190942 119149 pod_resize_test.go:1307] pod pod-resize-errors-1674/testpod running
> I0529 13:24:21.295785 119149 pod_resize_test.go:1205] pod pod-resize-test-6412/testpod running
> I0529 13:24:59.537041 119149 pod_resize_test.go:1205] pod pod-resize-test-3205/testpod running
> I0529 13:25:39.090500 119149 pod_resize_test.go:1205] pod pod-resize-test-9580/testpod running
> I0529 13:26:17.326333 119149 pod_resize_test.go:1205] pod pod-resize-test-5517/testpod running
> I0529 13:26:54.542900 119149 pod_resize_test.go:1205] pod pod-resize-test-5591/testpod running
> I0529 13:28:00.927663 119149 pod_resize_test.go:1205] pod pod-resize-test-5734/testpod running
> I0529 13:28:09.281611 119149 pod_resize_test.go:1205] pod pod-resize-test-5195/testpod running
> I0529 13:28:47.591446 119149 pod_resize_test.go:1205] pod pod-resize-test-8960/testpod running
> I0529 13:29:24.895025 119149 pod_resize_test.go:1205] pod pod-resize-test-6189/testpod running
> I0529 13:30:02.159139 119149 pod_resize_test.go:1205] pod pod-resize-test-6093/testpod running
> I0529 13:30:39.418359 119149 pod_resize_test.go:1205] pod pod-resize-test-2121/testpod running
> I0529 13:31:17.683862 119149 pod_resize_test.go:1205] pod pod-resize-test-1220/testpod running

Reply:
perhaps @tallclair meant use a new `f := framework.NewDefaultFramework("pod-resize-errors")` for each pod?